### DISCUSSION TITLE
Soliciting user feedback

### DISCUSSION THREAD

**Discussion by mundschenk-at** [astrandb/miele#38] [Ideas]
So I've been thinking about getting feedback from other users - when would you be comfortable to make this integration more well-known out there? Either by creating a new topic in the Community forums or adding a comment to the existing Miele@home topic.

I think it's already much more mature than the old integration, the only major issue at least for my devices is the program definition - i.e. making the sure it switches appropriately by device type. There are some other things that could be fine-tuned (or additional sensors to be added), but those are minor in comparison.

**Comment by astrandb** [comment]
I have had the same thoughts. 
The program labels should definitely be fixed so that we have a structure that supports all device types.
After that I think the integration is fairly complete with regard to the 4-5 types of appliances we own.
In order to support ovens, hoods etc new platforms should be added like lights and fans. They are fairly easy to create but we need users with access to those appliances for test and verification.

**Comment by mundschenk-at** [comment]
For hoods, I think we can do something based on the old integration - they recently added support for those platforms you mentioned and from the feedback posts, it looks like they worked reasonably well. Test and verification will still be needed of course, but we could start working on them before actual users start lining up.

**Reply by astrandb** [reply]
I have done fan and light platforms in other integrations so I know it not difficult at all. As long as the API behaves as expected.

Have you tried to setup and start programs from the integration? I suggest we leave to the user to call the raw service for the time beeing. I have a dishwasher from Siemens and the HA integration is a mess when it comes to programs. There is a switch entity for each and every possible program.

**Reply by mundschenk-at** [reply]
No I haven't tried it with the integration. I did it once with the official app, when the dishwasher was new, and it was cumbersome even there: You first have to set the device to "Mobile Start" and can then do the programming via the app. To me, that feels completely useless, as I have to fill the machines physically first anyway. What would be nice is being able to program and start the dishwasher from bed when you forgot to do it earlier, but that is not possible because unless it is specifically programmed (either with a later starting time or Mobile Start), the wifi module switches off to save energy.

So yeah, I agree leaving that for the raw service calls is fine for the foreseeable future.

**Reply by astrandb** [reply]
Another use case is to make an automation kick off the program when the electricity rate is low (or acceptable). Or to enable handling of load balancing between car chargers, saunas, water heaters etc.

**Reply by mundschenk-at** [reply]
Yes, but for that you could do the programming beforehand (with a late start time) and just start the machine "early" with the service. I'll check if that's possible with the official app. There's also the "Smart Grid" feature that I have not looked at closely because I don't have a time-based electricity rate (and no solar panels either). 

What I remember from researching these things is that the API is supposed to allow that, but does not actually work.

**Comment by mundschenk-at** [comment]
So are we good to go "public"?

**Reply by astrandb** [reply]
I wrote a reply to this topic earlier today but it disappeared somehow. I'll try to remember what I wrote...

We are getting close to going public
A few things should be addressed first:
 - Move i/o library to pypi - (PR ready)
 - Fix logging of program id
 - Improve handlig of extra_state_attributes
 - Perhaps add a comment in Readme.md where we explain why I started from scratch instead of liasing with the existing component developers.
None of these items are real showstoppers and we need inputs from a wider panel of testers so why not just go ahead.

Have you any suggestion how we should "advertise"

**Reply by mundschenk-at** [reply]
I haven't written a text yet, but I thought just an informal announcement:
* there's a new integration,
* its goal is to be aligned with HA best practices, 
* describe the current state (supported devices, potential rough edges)
* ask for user feedback in this repository.

**Comment by astrandb** [comment]
@mundschenk-at  Please try this automation. It might be good enough for tracing unknown programs etc
```yaml
alias: log_state_change
description: 'Write messages to HA log'
trigger:
  - platform: state
    entity_id: sensor.dishwasher_program
    attribute: Raw value
  - platform: state
    entity_id: sensor.dishwasher_program_phase
    attribute: Raw value
condition: []
action:
  - service: system_log.write
    data:
      message: '{{ trigger.entity_id }}: {{trigger.to_state.state }} | {{ trigger.to_state.attributes }}'
      level: warning
mode: single
```

**Reply by mundschenk-at** [reply]
Looks good. Not sure if it's the automation or the API not updating in time, but it seems I can switch through programs too fast for them to get picked up (I realized I don't have to actually start a program, selecting it is per se sufficient to change the sensor value).

**Reply by astrandb** [reply]
It is probably the API that keeps the pace. It pushes data whenever there is a change of any parameter. There might be some sort of throttling built-in though. 
I have converted the automation to a blueprint, see README.md. It needs some more love to be perfect, but it works.

**Reply by mundschenk-at** [reply]
Could the blueprint be expanded to support a list of entities?

**Reply by astrandb** [reply]
I have not found a way... But you can create one automation for each entity from the same blueprint.
There is a target selector that allows selection of multiple entities but it only works with actions and services, not with triggers.
It seems to be a limitation in blueprint functionality. I have seen many wishes for this feature in the forum.

**Reply by mundschenk-at** [reply]
I think the only thing that maybe works is manually entering the entities as a comma separated list. I'll just stay on the custom automation then, it's not like I need to change that very often.

**Comment by astrandb** [comment]
OK, time to publish...
https://community.home-assistant.io/t/miele-home-miele-mobile-component/64508/362
